sw/device/silicon_creator/manuf/base/ft_personalize.c

```diff
@@ -162,6 +189,31 @@ static status_t config_and_erase_certificate_flash_pages(void) {
   return OK_STATUS();
 }
 
+/**
+ * Helper function to compute measurements of various OTP partitions that are to
+ * be included in attestation certificates.
+ */
+static status_t measure_otp_partition(otp_partition_t partition,
+                                      hmac_digest_t *measurement) {
+  // Compute the digest.
+  otp_dai_read(partition, /*address=*/0, otp_state,
+               kOtpPartitions[partition].size / sizeof(uint32_t));
+
+  // Overwrites the CreatorSwCfg and OwnerSwCfg values read from OTP
+  // memory with the values from the OTP image. This is necessary because
+  // the CreatorSwCfg and OwnerSwCfg partitions are not fully provisioned
+  // in the OTP during the UDS certificate generation process.
+  if (partition == kOtpPartitionOwnerSwCfg) {
+    manuf_individualize_get_device_owner_sw_cfg((uint8_t *)otp_state);
```

Comment:
don't we only want to overwrite the AST values with something that is constant? like all ones?

additionally, the manuf_state field with its expected final value?

Whey do we read out the partition from OTP, then overwrite the whole thing instead of just the desired fields?

Reply:
> don't we only want to overwrite the AST values with something that is constant? like all ones?

If I understand correctly, the AST values should be already provisioned at the individualization stage (pulled from a flash info page). https://github.com/lowRISC/opentitan/blob/202127f06e209d0366eede40b39e59f40b54e559/sw/device/silicon_creator/manuf/lib/individualize_sw_cfg.c#L197

If we overwrite the AST values with a constant here, then the otp measurement calculated at this stage won't be matched with the final otp measurement written into the flash.

> additionally, the manuf_state field with its expected final value?

Right, that will be covered in #24789.

> Whey do we read out the partition from OTP, then overwrite the whole thing instead of just the desired fields?
I've updated the commit to explicitly update the desired fields, let me know if it looks good to you. My original thought was that explicitly overwriting the desired fields would require extra efforts to update a desired list when a new field needs to be provisioned at the end of perso process, while overwriting the whole thing wouldn't. And it should be safe to overwrite the whole partitions with the expected values for calculating the expected measurement because we will compare the OTP measurement used during certificate generation with the actual measurement stored in the OTP.

Reply:
Ah right r.e. AST values, they will already be programmed. Sorry I that is for another purpose I am thinking of.